File: Python/array.py
import random

def random_access(arr: list[int]) -> int:
    random_index = random.randint(0, len(arr) - 1)
    random_num = arr[random_index]
    return random_num


def insert(arr: list[int], index: int, value: int) -> None:
    for i in range(len(arr) - 1, index, -1):
        arr[i] = arr[i - 1]
    arr[index] = value


def remove(arr: list[int], index: int) -> None:
    for i in range(index, len(arr) - 1):
        arr[i] = arr[i + 1]


def find(arr: list[int], target: int) -> int:
    for i in range(len(arr)):
        if arr[i] == target:
            return i
    return -1

def extend(arr: list[int], enlarge: int) -> list[int]:
    res: list[int] = [0] * (len(arr) + enlarge)
    for i in range(len(arr)):
        res[i] = arr[i]
    return res

# test case
if __name__ == "__main__":
    arr = [1, 2, 3, 4, 5]
    # random access
    random_num = random_access(arr)
    print(random_num)

    # insert
    insert(arr, 1, 3)
    print(arr)

    # remove
    remove(arr, 3)
    print(arr)

    # find
    print(find(arr, 3))    

    # extend
    print(extend(arr, 2))

File: TS/src/array.ts
// const arr: number[] = [1, 2, 3, 4, 5];

// const randomNum = (arr: number[]): number => {
//   const randomIndex = Math.floor(Math.random() * arr.length);
//   return arr[randomIndex];
// };

// const insert = (arr: number[], index: number, value: number): void => {
//   for (let i = arr.length - 1; i > index; i--) {
//     arr[i] = arr[i - 1];
//   }
//   arr[index] = value;
// };

// const remove = (arr: number[], index: number): void => {
//   for (let i = index; i < arr.length - 1; i++) {
//     arr[i] = arr[i + 1];
//   }
// };

// const find = (arr: number[], target: number): number => {
//   for (let i = 0; i < arr.length; i++) {
//     if (arr[i] === target) {
//       return i;
//     }
//   }
//   return -1;
// };

// const extend = (arr: number[], enlarge: number): number[] => {
//   const res: number[] = new Array(arr.length + enlarge);
//   for (let i = 0; i < arr.length; i++) {
//     res[i] = arr[i];
//   }

//   return res;
// };

// //text case
// //random access
// console.log(randomNum(arr));
// //insert
// insert(arr, 1, 3);
// console.log(arr);

// //remove
// remove(arr, 3);
// console.log(arr);

// //find
// console.log(find(arr, 3));

// //extend
// console.log(extend(arr, 2));

// console.log(arr);
